File: engine/engine-model/src/main/scala/tools/packet/showitemgaininchat/ShowOwnBuffEffect.scala
package tools.packet.showitemgaininchat

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ShowOwnBuffEffect(private var _skillId: Int, private var _effectId: Int) extends PacketInput {
  def skillId: Int = _skillId

  def effectId: Int = _effectId

  override def opcode(): SendOpcode = SendOpcode.SHOW_ITEM_GAIN_INCHAT
}

File: engine/engine-model/src/main/java/tools/packet/showitemgaininchat/ShowOwnBuffEffect.java
package tools.packet.showitemgaininchat;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ShowOwnBuffEffect(Integer skillId, Integer effectId) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.SHOW_ITEM_GAIN_IN_CHAT;
   }
}